CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+Anyone is welcome to open a new issue, which is either related to a bug or a request for a new feature.
+
+### Reporting bugs
+
+In order to report a bug or misbehavior in urunc, a user can open a new issue explaining the problem.
+For the time being, we do not use any strict template for reporting any issues.
+However, in order to easily identify and fix the problem, it would be very helpful to provide enough information.
+In that context, when opening a new issue regarding a bug, we kindly ask you to:
+- Mark the issue with the bug label
+- Provide the following information:
+    1. A short description of the bug
+    2. Urunc's version (either the commit's hash or the version)
+    3. The CPU architecture, VMM and the Unikernel framework used.
+    4. Any particular steps to reproduce the issue.
+- Keep an eye on the issue for possible questions from the maintainers.
+
+A template for an issue could be the following one:
+```
+## Description
+An explanation of the issue 
+
+## System info
+- Urunc version:
+- Arch:
+- VMM:
+- Unikernel:
+
+## Steps to reproduce
+A list of steps that can reproduce the issue.
+```
+
+### Requesting new features
+
+We will be very happy to listen from users about new features that they would like to see in urunc.
+One way to communicate such a request is using Github issues.
+For the time being, we do not use any strict template for requesting new features.
+However, we kindly ask you to mark the issue with the enhancement label and provide a description of the new feature.
+
+## Submitting a PR
+
+Anyone should feel free to submit a change or an addition to the codebase of urunc.
+Currently, we use Github's Pull Requests (PRs) to submit changes to urunc's codebase.
+Before creating a new PR, please follow the below guidelines:
+- Make sure that the changes do not break the building process of urunc.
+- Make sure that the tests run successfully.
+- Use one commit for each new or changed functionality 
+- Make sure that no commit in a PR breaks the building process of urunc
+- Make sure to sign-off your commits.
+- Provide meaningful commit messages, describing shortly the changes the commit introduces.
+- Provide a meaningful PR message
+
+As soon as a new PR is created the following workflow will take place:
+1. One of urunc's maintainers will check the PR and invoke the tests, by adding the `ok-to-test` label.
+2. If the tests pass, one or more urunc's maintainers will review the PR.
+3. The author of the PR should address all the comments from the reviewers.
+4. As soon as a reviewer approves the PR, an action will add the appropriate git trailers in the PR's commits.
+5. The PR is now ready to merge.
+
+## Style guide
+
+### Git commit messages
+
+Please follow the below guidelines for your commit messages:
+- Limit the first line to 72 characters or less.
+- Limit all the other lines to 80 characters
+- In case the PR is associated with an issue, please refer to it.
```

Comment:
maybe the PR should be associated with an issue. Regardless, it should be mentioned in the commit message as a new line, before the `Signed-of-by` tag with:
```
Fixes: #Nr_issue
```